src/EditorFeatures/Core/Implementation/Adornments/GraphicsTag.cs

```diff
@@ -11,31 +13,53 @@ namespace Microsoft.CodeAnalysis.Editor.Implementation.Adornments
     /// </summary>
     internal abstract class GraphicsTag : ITag
     {
-        protected static SolidColorBrush VerticalRuleBrush;
-        protected static Color VerticalRuleColor;
+        private IEditorFormatMapService _editorFormatMapService;
+        private IEditorFormatMap _editorFormatMap;
 
-        protected virtual void Initialize(IWpfTextView view)
+        protected virtual void Initialize(IWpfTextView view, 
+            ref Brush graphicsTagBrush,
+            ref Color graphicsTagColor)
         {
-            if (VerticalRuleBrush != null)
+            if (graphicsTagBrush != null)
```

Comment:
This is a really gross pattern.  Can we rename this to something like EnsureInitialized and not do the ref to a field thing?